src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    currentState = { ...currentState };
```

Comment:
This line is redundant because you are copying `currentState` at the beginning of each iteration. Instead, you should make a copy only when you need to modify the state.